eng/source-build-release.yml

```diff
@@ -196,12 +201,32 @@ jobs:
       artifact: '$(TarballBuildArtifactLeg)'
       patterns: 'Private.SourceBuilt.Artifacts.$(SdkVersion).tar.gz'
 
+  - task: DownloadPipelineArtifact@2
+    name: DownloadSourceBuiltSDKStep
+    displayName: Download Source-Built SDK
+    inputs:
+      source: 'specific'
+      project: '$(azDoProject)'
+      pipeline: '$(installerTarabllBuildCiPipelineId)'
+      runVersion: 'specific'
+      runId: '$(InstallerTarballBuildRunId)'
+      artifact: '$(TarballBuildArtifactLeg)'
+      patterns: 'dotnet-sdk-$(SdkVersion)-*.tar.gz'
```

Comment:
[DownloadPipelineArtifact](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/download-pipeline-artifact-v2?view=azure-pipelines) allegedly supports multiple `patterns` but I couldn't get it to work. The syntax would be like:

```yaml
patterns: |
    'pattern1'
    'pattern2'
```